pandas/util/testing.py

```diff
@@ -515,9 +516,14 @@ def equalContents(arr1, arr2):
     return frozenset(arr1) == frozenset(arr2)
 
 
-def assert_index_equal(left, right, exact='equiv', check_names=True,
-                       check_less_precise=False, check_exact=True,
-                       check_categorical=True, obj='Index'):
+def assert_index_equal(left: Index,
+                       right: Index,
+                       exact: Union[bool, str] = 'equiv',
+                       check_names: bool = True,
+                       check_less_precise: Union[bool, int] = False,
+                       check_exact: bool = True,
+                       check_categorical: bool = True,
+                       obj: str = 'Index') -> None:
```

Comment:
Just a general comment but NoReturn may be applicable here at some point when we drop 3.5, though not sure if Optional[NoReturn] makes sense